src/plugins/navigation_embeddable/public/components/dashboard_link/dashboard_link_component.tsx

```diff
@@ -33,13 +39,10 @@ export const DashboardLinkComponent = ({
   const [error, setError] = useState<Error | undefined>();
 
   const dashboardContainer = navEmbeddable.parent as DashboardContainer;
-  const parentDashboardTitle = dashboardContainer.select((state) => state.explicitInput.title);
-  const parentDashboardDescription = dashboardContainer.select(
-    (state) => state.explicitInput.description
-  );
-
+  const parentDashboardInput = useObservable(dashboardContainer.getInput$());
```

Comment:
This fixes a React unmount state error where, on navigating to a different dashboard, the nav embeddable would try to update the state of the **previous** nav embeddable even though it was unmounted. Switching to the input observable prevents this.